src/content/api/plugins.mdx

```diff
@@ -87,18 +88,50 @@ The moral of the story is that there are a variety of ways to `hook` into the
 
 ## Custom Hooks
 
-In order to add a new hook to the compilation for other plugins to `tap` into,
-you then `require` the necessary hook class from `tapable` and create one:
+In order to offer a custom hook to the compilation for other plugins to `tap` into,
+you need to do the following:
 
-```js
-const SyncHook = require('tapable').SyncHook;
+- Create a module-scope `WeakMap` for compilation hooks:
 
-if (compiler.hooks.myCustomHook) throw new Error('Already in use');
-compiler.hooks.myCustomHook = new SyncHook(['a', 'b', 'c']);
+  ```ts
+  const compilationHooks = new WeakMap<Compilation, MyHooks>();
```

Comment:
Will it make more sense to use js instead of ts in this page?

Reply:
I feel it's ok to use typescript here as the usage is basic.